py/desispec/spectra.py

```diff
@@ -189,10 +203,10 @@ def __init__(self, bands=[], wave={}, flux={}, ivar={}, mask=None,
                 self.resolution_data[b] = resolution_data[b].astype(self._ftype)
                 self.R[b] = np.array( [ Resolution(r) for r in resolution_data[b] ] )
             if extra is not None:
-                self.extra[b] = {}
-                for ex in extra[b].items():
-                    self.extra[b][ex[0]] = np.copy(ex[1].astype(self._ftype))
-
+                if extra[b] is not None:
+                    self.extra[b] = {}
+                    for ex in extra[b].items():
+                        self.extra[b][ex[0]] = np.copy(ex[1].astype(self._ftype))
```

Comment:
Perhaps related to the previous comment: please explain the case that requires the additional `if extra[b] is not None` check.

Reply:
One could be even more stringent.  Since `extra` is a dictionary of dictionaries, do we insist that not only do all bands exist, but that every band dictionary has the same keys?

Again, since I didn't see that level of stringency in the original code, I assumed that the definition of `extra` could be "loose".